src/Symfony/Component/Form/Extension/Core/Type/WeekType.php

```diff
@@ -0,0 +1,189 @@
+    {
+        if ('single_text' === $options['widget']) {
+            $builder->addViewTransformer(new ReversedTransformer(new WeekDateToArrayTransformer()));
+            $builder->addModelTransformer(new WeekDateToArrayTransformer());
+        } else {
+            $yearOptions = $weekOptions = [
+                'error_bubbling' => true,
+                'empty_data' => '',
+            ];
+            // when the form is compound the entries of the array are ignored in favor of children data
+            // so we need to handle the cascade setting here
+            $emptyData = $builder->getEmptyData() ?: [];
+
+            $yearOptions['empty_data'] = $emptyData['year'] ?? '';
+            $weekOptions['empty_data'] = $emptyData['week'] ?? '';
+
+            if (isset($options['invalid_message'])) {
+                $yearOptions['invalid_message'] = $options['invalid_message'];
+                $weekOptions['invalid_message'] = $options['invalid_message'];
+            }
+
+            if (isset($options['invalid_message_parameters'])) {
+                $yearOptions['invalid_message_parameters'] = $options['invalid_message_parameters'];
+                $weekOptions['invalid_message_parameters'] = $options['invalid_message_parameters'];
+            }
+
+            if ('choice' === $options['widget']) {
+                // Only pass a subset of the options to children
+                $yearOptions['choices'] = array_combine($options['years'], $options['years']);
+                $yearOptions['placeholder'] = $options['placeholder']['year'];
+                $yearOptions['choice_translation_domain'] = $options['choice_translation_domain']['year'];
+
+                $weekOptions['choices'] = array_combine($options['weeks'], $options['weeks']);
+                $weekOptions['placeholder'] = $options['placeholder']['week'];
+                $weekOptions['choice_translation_domain'] = $options['choice_translation_domain']['week'];
+
+                // Append generic carry-along options
+                foreach (['required', 'translation_domain'] as $passOpt) {
+                    $yearOptions[$passOpt] = $options[$passOpt];
+                    $weekOptions[$passOpt] = $options[$passOpt];
+                }
+            }
+
+            $builder->add('year', self::$widgets[$options['widget']], $yearOptions);
+            $builder->add('week', self::$widgets[$options['widget']], $weekOptions);
+
+            $builder->addViewTransformer(new WeekDateToArrayTransformer());
+
+            if ('array' === $options['input']) {
+                $builder->addModelTransformer(new ReversedTransformer(new WeekDateToArrayTransformer()));
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        $view->vars['widget'] = $options['widget'];
+
+        if ($options['html5'] && 'single_text' === $options['widget']) {
```

Comment:
I think we should not allow to choose a different widget than `single_text` if `html5` is enabled for DX (we started to do the for the date related form types).